server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -470,6 +472,25 @@ protected void doMerge(final ObjectMapper mergeWith) {
         }
     }
 
+    private void throwExceptionIfEnabledAttributeIsUpdatedForType(ObjectMapper mergeWith, Mapper mergeWithMapper, Mapper mergeIntoMapper) {
+        if (mergeIntoMapper instanceof ObjectMapper && mergeWithMapper instanceof ObjectMapper) {
+            final ObjectMapper mergeIntoObjectMapper = (ObjectMapper) mergeIntoMapper;
+            final ObjectMapper mergeWithObjectMapper = (ObjectMapper) mergeWithMapper;
+
+            if (mergeIntoObjectMapper.isEnabled() != mergeWithObjectMapper.isEnabled()) {
+                if (mergeIntoObjectMapper.nestedTypeFilter() instanceof TermQuery) {
```

Comment:
I'm not quite sure I understand what the `enable` setting in mappings that are merged have to do with nested queries, especially with TermQuery. Maybe I'm missing sth., could you explain?

Reply:
I dug into this with a debugger and the `_type` attribute happened to be wrapped into a `TermQuery`.

I've been a (tiny) bit paranoiac as I wanted to be sure to not make other things break.

I'll have another debugging session to see how things work when updating mapping with `"_source": { "enabled": false }`. I'll let you know.

Reply:
Well, it looks like updating the `_source` attribute in a mapping is not allowed.

**command**

```
curl -XPUT "${ES_CLUSTER}/index1/_mapping/type1" -H "Content-Type: application/json" --data '{
  "properties": {
    "foo": {
      "source": {
        "enabled": false
      },
      "properties": {
        "bar": {
          "type": "text"
        }
      }
    }
  }
}'
```

**output**

```
{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "Mapping definition for [foo] has unsupported parameters:  [source : {enabled=false}]"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "Mapping definition for [foo] has unsupported parameters:  [source : {enabled=false}]"
  },
  "status": 400
}
```

I'll now have a look if removing this conditional block doesn't conflict with other `enabled`-like metadata attributes.